src/main/java/org/wildfly/security/auth/provider/FileSystemSecurityRealm.java

```diff
@@ -469,7 +472,16 @@ private void writeIdentity(final XMLStreamWriter streamWriter, final LoadedIdent
                 do {
                     streamWriter.writeCharacters("\n        ");
                     final Object credential = credIter.next();
-                    if (credential instanceof Password) {
+                    if (credential instanceof OneTimePassword) {
+                        final OneTimePassword otp = (OneTimePassword) credential;
+                        otp.getHash();
+                        streamWriter.writeStartElement(ELYTRON_1_0, "otp");
+                        streamWriter.writeAttribute("algorithm", otp.getAlgorithm());
+                        streamWriter.writeAttribute("hash", ByteIterator.ofBytes(otp.getHash()).hexEncode().drainToString());
+                        streamWriter.writeAttribute("seed", new String(otp.getSeed(), StandardCharsets.US_ASCII));
```

Comment:
Is this going to be an ASCII string, always?  Maybe base64 is a better choice here too?